File: Chapter06/Advanced-Microservices/src/main/java/com/mastering/spring/springboot/controller/BasicController.java
package com.mastering.spring.springboot.controller;

import java.util.Locale;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.MessageSource;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestHeader;
import org.springframework.web.bind.annotation.RestController;

import com.mastering.spring.springboot.bean.WelcomeBean;

@RestController
public class BasicController {

	@Autowired
	private MessageSource messageSource;

	@GetMapping("/welcome")
	public String welcome() {
		return "Hello World";
	}

	@GetMapping("/welcome-with-object")
	public WelcomeBean welcomeWithObject() {
		return new WelcomeBean("Hello World");
	}

	private static final String helloWorldTemplate = "Hello World, %s!";

	@GetMapping("/welcome-with-parameter/name/{name}")
	public WelcomeBean welcomeWithParameter(@PathVariable String name) {
		return new WelcomeBean(String.format(helloWorldTemplate, name));
	}

	@GetMapping("/welcome-internationalized")
	public String msg(@RequestHeader(value = "Accept-Language", required = false) Locale locale) {
		return messageSource.getMessage("welcome.message", null, locale);
	}
}

File: Chapter13/first-web-service-with-kotlin/src/main/kotlin/com/mastering/spring/kotlin/firstwebservicewithkotlin/controller/BasicController.kt
package com.mastering.spring.kotlin.firstwebservicewithkotlin.controller

import org.springframework.web.bind.annotation.RestController
import org.springframework.web.bind.annotation.GetMapping
import com.mastering.spring.kotlin.firstwebservicewithkotlin.bean.WelcomeBean
import org.springframework.web.bind.annotation.PathVariable

@RestController
class BasicController {
	@GetMapping("/welcome")
	fun welcome() = "Hello World"

	@GetMapping("/welcome-with-object")
	fun welcomeWithObject() = WelcomeBean("Hello World")

	@GetMapping("/welcome-with-parameter/name/{name}")
	fun welcomeWithParameter(@PathVariable name: String) = WelcomeBean("Hello World, $name")
}